client/src/components/SelectedCourse/SelectedCourse.js

```diff
@@ -1,23 +1,29 @@
-import React from "react";
+import React from 'react';
 import './SelectedCourse.css';
 
 import {
-  Link
-} from "react-router-dom";
+  Link,
+} from 'react-router-dom';
 
-function SelectedCourse({id, title, headline, url, image,
- num_lectures, num_subscribers}){
+function SelectedCourse({ state, setState,
+}) { 
+  window.onbeforeunload = () => {
+    setState({ ...state, selectedCourse: null });
+  };
```

Comment:
instead of using this you should use the return statement from useEffect, since the return statement runs before the component is unmounted